Common/Data/Custom/PsychSignal/PsychSignalSentimentData.cs

```diff
@@ -0,0 +1,137 @@
+        /// This is the total nubmer of messages classified as bullish in a minute
+        /// </summary>
+        public int BullScoredMessages;
+
+        /// <summary>
+        /// Total bearish scored messages. 
+        /// This is the total number of messages classified as bearish in a minute
+        /// </summary>
+        public int BearScoredMessages;
+
+        /// <summary>
+        /// Bull/Bear message ratio.
+        /// Calculated by dividing BullScoredMessages by BearScoredMessages
+        /// </summary>
+        /// <remarks>If bearish messages equals zero, then the resulting value equals zero</remarks>
+        public decimal BullBearMessageRatio;
+
+        /// <summary>
+        /// Total messages scanned.
+        /// This includes bull/bear messages and messages that couldn't be classified
+        /// </summary>
+        /// <remarks>
+        /// Sometimes, there will be no bull/bear rated messages, but nonetheless had messages scanned.
+        /// This field describes the total fields that were scanned in a minute
+        /// </remarks>
+        public int TotalScoredMessages;
+
+        /// <summary>
+        /// Retrieve Psychsignal data from disk and return it to user's custom data subscription
+        /// </summary>
+        /// <param name="config">Configuration</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in livemode, false for backtesting mode</param>
+        /// <returns></returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "alternative",
+                    "psychsignal",
+                    $"{config.Symbol.Value.ToLower()}",
+                    $"{date:yyyyMMdd}.zip#{date:yyyyMMdd}.csv"
+                ),
+                SubscriptionTransportMedium.LocalFile,
+                FileFormat.Csv
+            );
+        }
+
+        /// <summary>
+        /// Reads a single entry from psychsignal's data source.
+        /// </summary>
+        /// <param name="config">Subscription data config setup object</param>
+        /// <param name="line">Line of the source document</param>
+        /// <param name="date">Date of the requested data</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        ///     Instance of the T:BaseData object containing psychsignal specific data
+        /// </returns> 
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            try
+            {
+                var csv = line.Split(',');
+
+                var timestamp = new DateTime(date.Year, date.Month, date.Day).AddMilliseconds(
```

Comment:
I believe you could directly create the new `DateTime` instance with the milliseconds specified, avoiding an extra `DateTime` instance creation

Reply:
yup, should be able to do:
``` csharp
var timestamp = date.AddMilliseconds(Convert.ToDouble(csv[0]));
```

Reply:
also, do we want to add `CultureInfo.InvariantCulture` to ensure compatibility?